error-handler.js

```diff
@@ -98,15 +99,16 @@ var mixIn = require('mout/object/mixIn'),
     stream.pipe(res);
   },
 
-  send = function send(statusCode, err, res, o) {
+  send = function send(statusCode, err, res, o) {    
+    if (!err.message) err.message = statusCodes[statusCode];
```

Comment:
Who's idea was it to make basic properties of the error object non-enumerable in the first place? =)

> why is it improper to mutate the err object at this point?

`err` is passed as a reference. We didn't create the object, and we don't know if/how it may be used elsewhere (short of digging into the source code to find out, and the source code could change in later versions).

If we mutate it, we could break code we don't even know about. Maybe it's accessed during the `server.close()` operation? Keep in mind, this lib works in two different frameworks, so double that danger.

>  Is anything else going to inspect the err object after we handle this error?

Since we don't own the object, the answer to this question is unknowable. That's the trouble with shared mutable state...

> Moreover, we are adding a few properties only if they are falsy ('' for err.message and undefined for anything else), not overwriting anything. Are we mutating the error object, or possibly just augmenting it as we handle it?

Because we can't examine code we don't know about, we have no idea how the mutation might impact that code... Sure, I agree that deleting an expected property is more likely to cause problems than adding properties, but if there is any code checking for example, the length of `Object.keys()` (admittedly, a silly, brittle thing to do), it would break that code.

But since this lib is used in high-availability production applications with millions of monthly active users... this mutation could cause costly problems.